nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,46 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+// 	Respond with the json for all the meals and  /api/meals
+
+router.get("/", async (req, res) => {
+
+  if( (req.query.maxPrice && isNaN(parseInt(req.query.maxPrice))) || 
+      (req.query.createdAfter  && isNaN(Date.parse(req.query.createdAfter))) ||
+      (req.query.limit && isNaN(parseInt(req.query.limit)))
+    ){
+    return res.status(400).send({"error": "Parameter not Acceptable"})
+  }
+
+    const filteredData = meals.filter( (meal) => req.query.maxPrice ? meal.price < req.query.maxPrice: true) 
+    .filter((meal) => req.query.title? meal.title.toLowerCase().includes(req.query.title.toLowerCase()) : true)
+    .filter((meal) => req.query.createdAfter ? Date.parse(meal.createdAt) > Date.parse(req.query.createdAfter) : true)
+    
+    if (req.query.limit && isNaN(parseInt(req.query.limit))){
```

Comment:
Here you need to add '!' to isNaN(parseInt(req.query.limit). After that, you will see, that there is an error here, since you are trying to assign something for a constant (conts) variable).

Reply:
I fixed it